src/main/java/org/ecocompass/core/graph/Graph.java

```diff
@@ -42,6 +43,102 @@ private double calculateDistance(Node node1, Node node2) {
         double c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
         return 6371 * c;
     }
+
+    public List<Node> shortestPath(Long startNodeId, Long endNodeId, String transportType) throws Exception {
```

Comment:
We should keep separate utility classes for all these functions. That will make this file much cleaner and more understandable.